README.md

```diff
@@ -0,0 +1,303 @@
+- `middleware: (ctx: Object, next: () => Promise) => Promise` - Optional. A Koa middleware
+- `plugin: {Service, middleware, of}`
+  - `Service: class`
+  - `middleware: (ctx: Object, next: () => Promise) => Promise`
+  - `of: (ctx: Object|null|undefined) => Service` - returns an instance of the service, memoized using `ctx` as the memoization key.
+
+#### Context
+
+Middlewares receive a `ctx` object as their first argument. This object has a property called `element` in both server and client.
+
+In the server, `ctx` also exposes the same properties as a [Koa context](http://koajs.com/#context)
+
+Additionally, when server-side rendering a page, FusionJS sets `ctx.body` to an object with the following properties:
+
+- `htmlAttrs: Object` - attributes for the `<html>` tag. For example `{lang: 'en-US'}` turns into `<html lang="en-US">`. Default: empty object
+- `title: string` - The content for the `<title>` tag. Default: empty string
+- `head: Array` - A list of [sanitized HTML strings](#html-sanitization). Default: empty array
+- `body: Array` - A list of [sanitized HTML strings](#html-sanitization). Default: empty array
+- `ssr: string` - When the virtual dom `render` is called, this property is populated with the rendered html string
+
+When a request does not require a server-side render, `ctx.body` follows regular Koa semantics.
+
+#### HTML sanitization
+
+Default-on HTML sanitization is important for preventing security threats such as XSS attacks.
+
+Fusion automatically sanitizes `htmlAttrs` and `title`. When pushing HTML strings to `head` or `body`, you must use the `html` template tag to mark your HTML as sanitized:
+
+```js
+import {html} from 'fusion-core';
+
+export default () => (ctx, next) => {
+  if (ctx.element) {
+    const userData = await getUserData();
+    // userData can't be trusted, and is automatically escaped
+    ctx.body.body.push(html`<div>${userData}</div>`)
+  }
+  return next();
+}
+```
+
+If `userData` above was `<script>alert(1)</script>`, the string would be automatically turned into `<div>&lt;script&gt;alert(1)&lt;/script&gt;</div>`. Note that only `userData` is escaped, but the HTML in your code stays intact.
+
+If your HTML is complex and needs to be broken into smaller strings, you can also nest sanitized HTML strings like this:
+
+```js
+const notUserData = html`<h1>Hello</h1>`
+const body = html`<div>${notUserData}</div>`
+```
+
+Note that you cannot mix sanitized HTML with unsanitized strings:
+
+```js
+ctx.body.body.push(html`<h1>Safe</h1>` + 'not safe') // will throw an error when rendered
+```
+
+Also note that only template strings can have template tags (i.e. <code>html&#x60;&lt;div&gt;&lt;/div&gt;&#x60;</code>). The following are NOT valid Javascript: `html"<div></div>"` and `html'<div></div>'`.
+
+If you get an <code>Unsanitized html. You must use html&#x60;[your html here]&#x60;</code> error, remember to prepend the `html` template tag to your template string.
```

Comment:
This also I think is not in sync

Reply:
No, this is fine. It's a quirk of markdown syntax that we need to html-escape backticks and angled brackets when we want to actually them print out

Reply:
Ah, got it 👍